server/routes/analyticsRouter.ts

```diff
@@ -1,10 +1,21 @@
-import type { RequestHandler, Router } from 'express'
+import type { NextFunction, Request, Response, RequestHandler, Router } from 'express'
+import { NotFound } from 'http-errors'
 
 import asyncMiddleware from '../middleware/asyncMiddleware'
 import AnalyticsService from '../services/analyticsService'
 
 export default function routes(router: Router): Router {
-  const get = (path: string, handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
+  const featureGate = (handler: RequestHandler): RequestHandler => {
+    return (req: Request, res: Response, next: NextFunction): void => {
+      if (req.app.locals.featureFlags.showAnalytics) {
+        handler(req, res, next)
+      } else {
+        next(new NotFound())
+      }
+    }
+  }
```

Comment:
Wouldn't be easier to add a similar `if` statement in the `routes/all.ts` file? effect should be the same right?

Reply:
do you mean so that the analytics router isn't added to the parent? it wouldn't be possible to test the feature gate cos the router tree is only setup once, right?

Reply:
aah actually: conditionally adding this router in `routes/all.ts` doesn't work for a different reason. there's no way i can see to reach the `app.locals` object in that parent router without wrapping in a request handler that'd do basically the same thing as this.

i think i'll make this feature gate a shareable utility though